libtenzir/builtins/operators/deduplicate.cpp

```diff
@@ -277,9 +403,11 @@ class plugin final : public virtual operator_plugin<deduplicate_operator> {
     std::optional<int64_t> limit{};
     std::optional<int64_t> distance{};
     std::optional<duration> timeout{};
+    bool project_only{false};
     parser.add("--limit", limit, "<count>");
     parser.add("--distance", distance, "<count>");
     parser.add("--timeout", timeout, "<duration>");
+    parser.add("--project-only", project_only);
```

Comment:
This isn't really undocumented as listed in the PR description, as it shows up when trying to parse a wrong option. I.e., `deduplicate --help` will list this option. Do we have a good solution for this?

cc @jachris

Reply:
I could just rip it out, but it was really convenient when developing this PR, and I could imagine it being useful for development in the future

Reply:
We could add a parameter to make it hidden.

Reply:
I don't feel too strongly about it and we can also leave it in.

Reply:
I'll just leave it in, then.